lib/Doctrine/Migrations/Provider/SchemaDiffProviderInterface.php

```diff
@@ -6,6 +6,9 @@
 
 use Doctrine\DBAL\Schema\Schema;
 
+/**
+ * @internal
```

Comment:
isn't this meant to be an extension point ?

Reply:
No, only `SchemaProviderInterface` is an extension point that can be passed through here https://github.com/doctrine/migrations/blob/master/lib/Doctrine/Migrations/Tools/Console/Command/DiffCommand.php

I think the interface might have been created to ease unit testing whenever it was added.

Reply:
the interface has been added because the lazy one is implemented through decoration

Reply:
Understood. So I think this should not be public still, right?

Reply:
yeah, this one is OK. I confused it with `SchemaProviderInterface`